radix-engine/src/kernel/kernel.rs

```diff
@@ -766,31 +798,24 @@ where
         &mut self,
         lock_handle: LockHandle,
     ) -> Result<&IndexedScryptoValue, RuntimeError> {
-        let mut read_from_heap = false;
         let mut handler = KernelHandler {
             callback: self.callback,
             prev_frame: self.prev_frame_stack.last(),
-            on_store_access: |_, store_access| {
-                read_from_heap = matches!(store_access, StoreAccess::ReadFromHeap);
-                Ok(())
+            on_store_access: |_, _| {
```

Comment:
@talekhinezh maybe we should change that field to `Option<F>`, so here we can just pass `None` and in other places `Some(...)`?

Reply:
But I see there is a complication of using `None` with `Option<F>`.